File: order-domain/src/main/java/com/order/vo/Nickname.java
package com.order.vo;

public record Nickname(String value) {

    public Nickname {
        if (value == null || value.isBlank()) {
            throw new IllegalArgumentException("Nickname cannot be blank");
        }
        if (value.length() > 20) {
            throw new IllegalArgumentException("Nickname cannot have more than 20 characters");
        }
    }
}

File: order-domain/src/main/kotlin/com/order/vo/Nickname.kt
package com.order.vo

@JvmInline
value class Nickname(
    val value: String,
) {
    init {
        require(value.isNotBlank()) { "Nickname cannot be blank" }
        require(value.length <= 20) { "Nickname cannot have 20 characters" }
    }
}